app/controllers/provider_interface/notifications_controller.rb

```diff
@@ -1,11 +1,14 @@
 module ProviderInterface
   class NotificationsController < ProviderInterfaceController
     def update
-      current_provider_user.update!(
-        send_notifications: notification_params[:send_notifications],
-      )
-
       if FeatureFlag.active?(:configurable_provider_notifications)
+        notification_preferences = current_provider_user.notification_preferences
+        notification_preferences.assign_attributes(notification_preferences_params)
+        notification_preferences.save!
```

Comment:
I think that should also work?

```suggestion
        notification_preferences.update!(notification_preferences_params)
```